server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -3217,7 +3216,7 @@ public void recover(VolumeManager fs, KeyExtent extent, List<LogEntry> logEntrie
     Collections.sort(sorted, (e1, e2) -> (int) (e1.timestamp - e2.timestamp));
     for (LogEntry entry : sorted) {
       Path recovery = null;
-      Path finished = RecoveryPath.getRecoveryPath(fs.getFullPath(FileType.WAL, entry.filename));
+      Path finished = RecoveryPath.getRecoveryPath(new Path(entry.filename));
```

Comment:
Should this be validated?

Reply:
I am not sure what we could validate with the WAL filenames.  But the filename is coming from LogEntry so I think we should do it when the LogEntry is created, not after the TabletServer gets the list of files.

Reply:
> so I think we should do it when the LogEntry is created

I like that approach.  We do need to avoid constantly validating the same data over and over in the code.  I think the goal should be that data is only validated after its read from a persistent store and no where else.

Reply:
I checked and ```LogEntry.fromKeyValue(key, val)``` does some checking. And this is what is called when the data is read from the metadata in ```TabletMetadata```